src/server/src/main/java/io/cassandrareaper/resources/RepairScheduleResource.java

```diff
@@ -76,6 +75,160 @@ public RepairScheduleResource(AppContext context) {
+          Integer scheduleDaysBetween,
+          Integer segmentCountPerNode
+  ) {
+    List<String> invalidParams = Lists.newArrayList();
+
+    boolean ownerValid = owner != null && !owner.isEmpty();
+
+    boolean repairParallelismValid = repairParallelism != null && !repairParallelism.isEmpty();
+    if (repairParallelismValid) {
+      try {
+        repairParallelismValid = RepairParallelism.valueOf(repairParallelism.toUpperCase()) != null ? true : false;
+      } catch (Exception e) {
+        repairParallelismValid = false;
+      }
+    }
+    boolean intensityValid = intensity != null && intensity >= 0.0D && intensity <= 1.0D;
+    // TODO What are good valid ranges for these values?
+    boolean scheduleDaysBetweenValid = scheduleDaysBetween != null && scheduleDaysBetween > 0;
+    boolean segmentCountPerNodeValid = segmentCountPerNode != null && segmentCountPerNode > 0;
+
+    if (!ownerValid) {
+      invalidParams.add("owner");
+    }
+    if (!repairParallelismValid) {
+      invalidParams.add("repairParallelism");
+    }
+    if (!intensityValid) {
+      invalidParams.add("intensity");
+    }
+    if (!scheduleDaysBetweenValid) {
+      invalidParams.add("scheduleDaysBetween");
+    }
+    if (!segmentCountPerNodeValid) {
+      invalidParams.add("segmentCountPerNode");
+    }
+
+    return invalidParams;
+  }
+
+  /**
+   * Utility method to apply any valid parameters to an existing RepairSchedule.
+   * This method assumes that any non-null parameter provided is valid and should
+   * be applied.
+   *
+   * @param repairSchedule
+   * @param owner
+   * @param repairParallelism
+   * @param intensity
+   * @param scheduleDaysBetween
+   * @param segmentCountPerNode
+   */
+  protected static RepairSchedule applyRepairPatchParams(
+          RepairSchedule repairSchedule,
+          String owner,
+          String repairParallelism,
+          Double intensity,
+          Integer scheduleDaysBetween,
+          Integer segmentCountPerNode
+  ) {
+    if (repairSchedule == null) {
+      return null;
+    }
+
+    // Apply any valid incoming values to the schedule
+    return repairSchedule.with()
+            .owner(owner != null ? owner.trim() : repairSchedule.getOwner())
+            .repairParallelism(repairParallelism != null ? RepairParallelism.valueOf(repairParallelism.toUpperCase()) : repairSchedule.getRepairParallelism())
+            .intensity(intensity != null ? intensity : repairSchedule.getIntensity())
+            .daysBetween(scheduleDaysBetween != null ? scheduleDaysBetween : repairSchedule.getDaysBetween())
+            .segmentCountPerNode(segmentCountPerNode != null ? segmentCountPerNode : repairSchedule.getSegmentCountPerNode())
+            .build(repairSchedule.getId());
+  }
+
+  @PATCH
+  @Path("/{id}")
+  public Response patchRepairSchedule(
+          @Context UriInfo uriInfo,
+          @PathParam("id") UUID repairScheduleId,
+          @QueryParam("owner") Optional<String> owner,
```

Comment:
@adejanovski I don't love passing these as query parameters and not as a resource in the body, but this is what was done for the PUT request that is currently used to update status, so I followed that model - was what was done for the other requests preferred or would you want to break from the convention?

Reply:
If the REST convention is to pass them in the body then I'm happy for you to do it, even if it doesn't follow what we've done elsewhere in the code (as any other OSS project, it's made of many peeps knowledge or lack of, so it's not necessarily done the right way).

Reply:
Cool, I went ahead and made the change.  It has the other added benefit of simplifying the validation pieces and lets Dropwizard handle it by itself with the addition of a few custom validators.

it simplifies the resource code and you also get some nicer error responses from Dropwizard that can be useful, like:

```
PATCH http://127.0.0.1:8080/repair_schedule/da706f30-0c3e-11ec-be46-915dd6e16900
{
    "owner": "",
    "intensity": 0.5,
    "repair_parallelism": "DATACENTER_AWARE",
    "scheduled_days_between": 60,
    "segment_count_per_node": -10
}
```

```
422
{
    "errors": [
        "segmentCountPerNode must be greater than or equal to 1",
        "daysBetween must be less than or equal to 31",
        "owner must not be empty if it is provided"
    ]
}
```

Dropwizard provides that response automatically based on the validation rules configured for the class.